c/blake3_neon.c

```diff
@@ -1,5 +1,5 @@
 #include "blake3_impl.h"
-
+#if !defined(_WIN32)
```

Comment:
Why this change? Anything not explicitly targeting ARM NEON should be skipping this file entirely.

Reply:
Guarding content of blake3_neon.c is recall/same approach used at request about ['using C code in C++'](https://github.com/BLAKE3-team/BLAKE3/pull/40). But in this case we have choose:
- even do not include this source file, through CMakeLists.txt, Visual Studio project, what ever other thing that able to build source files;
- or just guard file via macros - so no mater how we create project in this case.
It is no problem to not include this macros, because comparing to "external 'C'" construction such approach not so popular.

In case of "external 'C'" another approach will be that C++ side include C file inside proposed construction. 
```
extern "C" {
#include "blake3.h"
};
```